platform/qt/src/qmapboxgl.cpp

```diff
@@ -749,99 +833,227 @@ mbgl::Annotation fromPointAnnotation(const PointAnnotation &pointAnnotation) {
     return mbgl::SymbolAnnotation { mbgl::Point<double> { coordinate.second, coordinate.first }, icon.toStdString() };
 }
 
-AnnotationID QMapboxGL::addPointAnnotation(const PointAnnotation &pointAnnotation)
+/*!
+    Adds a \a point annotation to the map.
+
+    Returns the unique identifier for the new annotation.
+
+    \sa addAnnotationIcon()
+*/
+QMapbox::AnnotationID QMapboxGL::addPointAnnotation(const QMapbox::PointAnnotation &point)
 {
-    return d_ptr->mapObj->addAnnotation(fromPointAnnotation(pointAnnotation));
+    return d_ptr->mapObj->addAnnotation(fromPointAnnotation(point));
 }
 
-void QMapboxGL::updatePointAnnotation(AnnotationID id, const PointAnnotation &pointAnnotation)
+/*!
+    Updates an existing \a point annotation referred by \a id.
+
+    \sa addAnnotationIcon()
+*/
+void QMapboxGL::updatePointAnnotation(QMapbox::AnnotationID id, const QMapbox::PointAnnotation &point)
 {
-    d_ptr->mapObj->updateAnnotation(id, fromPointAnnotation(pointAnnotation));
+    d_ptr->mapObj->updateAnnotation(id, fromPointAnnotation(point));
 }
 
-AnnotationID QMapboxGL::addShapeAnnotation(const ShapeAnnotation &shapeAnnotation)
+/*!
+    Adds a \a shape annotation to the map.
+
+    Returns the unique identifier for the new annotation.
+*/
+QMapbox::AnnotationID QMapboxGL::addShapeAnnotation(const QMapbox::ShapeAnnotation &shape)
 {
-    return d_ptr->mapObj->addAnnotation(fromQMapboxGLShapeAnnotation(shapeAnnotation));
+    return d_ptr->mapObj->addAnnotation(fromQMapboxGLShapeAnnotation(shape));
 }
 
-void QMapboxGL::removeAnnotation(AnnotationID annotationID)
+/*!
+    Removes an existing annotation referred by \a id.
+*/
+void QMapboxGL::removeAnnotation(QMapbox::AnnotationID id)
 {
-    d_ptr->mapObj->removeAnnotation(annotationID);
+    d_ptr->mapObj->removeAnnotation(id);
 }
 
-void QMapboxGL::setLayoutProperty(const QString& layer_, const QString& property, const QVariant& value)
+/*!
+    Sets a layout \a property \a value to an existing \a layer. The \a property string can be any
+    as defined by the \l {https://www.mapbox.com/mapbox-gl-style-spec/} {Mapbox style specification}
+    for layout properties.
+
+    This example hides the layer \c route:
+
+    \code
+        map->setLayoutProperty("route", "visibility", "none");
+    \endcode
+
+    This table describes the mapping between \l {https://www.mapbox.com/mapbox-gl-style-spec/#types}
+    {style types} and Qt types accepted by setLayoutProperty():
```

Comment:
This table also applies to `setPaintProperty()`, so copy this table to that method’s documentation. (In particular, color-typed properties are paint properties.)